File: StillHereHope/Views/Screens/YouScreen.swift
import SwiftUI

struct YouView: View {
    @StateObject var viewModel: YouViewModel

    var body: some View {
        AppScreen(title: "You", subtitle: "A quiet view of what has helped and what you’ve carried.") {
            statsGrid
            impactCard
            moodHistorySection

            NavigationLink {
                SavedNotesView(notes: viewModel.savedNotes)
            } label: {
                QuietActionRow(title: "Saved notes", systemImage: "bookmark.fill")
            }
            .buttonStyle(.plain)

            NavigationLink {
                SettingsView()
            } label: {
                QuietActionRow(title: "Settings", systemImage: "gearshape.fill")
            }
            .buttonStyle(.plain)

            NavigationLink {
                ReportConcernView()
            } label: {
                QuietActionRow(title: "Report a concern", systemImage: "flag.fill")
            }
            .buttonStyle(.plain)
        }
        .navigationTitle("You")
        .navigationBarTitleDisplayMode(.inline)
    }

    private var statsGrid: some View {
        HStack(alignment: .top, spacing: AppTheme.Spacing.small) {
            StatCard(title: "day streak", value: "\(viewModel.stats.streakCount)", systemImage: "flame")
            StatCard(title: "saved notes", value: "\(viewModel.stats.totalSavedNotes)", systemImage: "bookmark")
        }
    }

    private var impactCard: some View {
        VStack(alignment: .leading, spacing: AppTheme.Spacing.xxSmall) {
            Text("Impact".uppercased())
                .font(.caption.weight(.semibold))
                .tracking(0.8)
                .foregroundStyle(AppTheme.Colors.textTertiary)
            Text("Your notes helped \(viewModel.stats.totalHelpfulNotes) people")
                .font(.title3.weight(.semibold))
                .foregroundStyle(AppTheme.Colors.textPrimary)
            Text("Even short words can help someone keep going.")
                .font(.body)
                .foregroundStyle(AppTheme.Colors.textSecondary)
        }
        .hopeCardStyle()
    }

    private var moodHistorySection: some View {
        VStack(alignment: .leading, spacing: AppTheme.Spacing.small) {
            SectionHeader(
                eyebrow: "Patterns",
                title: "Mood history",
                subtitle: "A soft record of what days have asked from you."
            )

            if viewModel.moodHistory.isEmpty {
                EmptyStateCard(
                    title: "No check-ins yet",
                    message: "Your check-ins will gently collect here so you can notice patterns without pressure.",
                    systemImage: "chart.bar"
                )
            } else {
                VStack(spacing: AppTheme.Spacing.small) {
                    ForEach(viewModel.moodHistory) { checkIn in
                        HStack(spacing: AppTheme.Spacing.small) {
                            Image(systemName: checkIn.mood.symbol)
                                .foregroundStyle(AppTheme.Colors.accent)
                                .frame(width: 26)
                            VStack(alignment: .leading, spacing: 2) {
                                Text(checkIn.mood.title)
                                    .font(.headline)
                                    .foregroundStyle(AppTheme.Colors.textPrimary)
                                Text(checkIn.suggestedAction)
                                    .font(.subheadline)
                                    .foregroundStyle(AppTheme.Colors.textSecondary)
                                    .lineLimit(2)
                            }
                            Spacer()
                            Text(checkIn.date.formatted(date: .abbreviated, time: .omitted))
                                .font(.caption)
                                .foregroundStyle(AppTheme.Colors.textSecondary)
                        }
                        .padding(AppTheme.Spacing.small)
                        .background(AppTheme.Colors.cardStrong)
                        .clipShape(RoundedRectangle(cornerRadius: AppTheme.Corners.button, style: .continuous))
                        .overlay(
                            RoundedRectangle(cornerRadius: AppTheme.Corners.button, style: .continuous)
                                .stroke(AppTheme.Colors.border, lineWidth: 1)
                        )
                    }
                }
                .hopeCardStyle(padding: AppTheme.Spacing.small)
            }
        }
    }
}

#Preview("You") {
    NavigationStack {
        YouView(viewModel: YouViewModel(repository: HopeRepository()))
    }
}

#Preview("You - iPad", traits: .fixedLayout(width: 820, height: 1180)) {
    NavigationStack {
        YouView(viewModel: YouViewModel(repository: HopeRepository()))
    }
}

File: still_here_hope_flutter/lib/features/you/you_screen.dart
import 'package:flutter/material.dart';
import 'package:flutter_riverpod/flutter_riverpod.dart';
import 'package:go_router/go_router.dart';

import '../../models/app_models.dart';
import '../../views/components/calm_components.dart';
import '../notes/data/hope_repository.dart';

class YouScreen extends ConsumerWidget {
  const YouScreen({super.key});

  @override
  Widget build(BuildContext context, WidgetRef ref) {
    final repo = ref.watch(hopeRepositoryProvider);
    final stats = repo.stats;

    return Scaffold(
      body: CalmScreen(
        title: 'You',
        subtitle: 'A quiet view of what has helped and what you have carried.',
        children: [
          LayoutBuilder(
            builder: (context, constraints) {
              final wide = constraints.maxWidth > 520;
              final cards = [
                _StatCard(icon: Icons.local_fire_department_outlined, value: '${stats.streakCount}', label: 'day streak'),
                _StatCard(icon: Icons.bookmark_border, value: '${stats.totalSavedNotes}', label: 'saved notes'),
              ];
              return wide
                  ? Row(children: cards.map((card) => Expanded(child: Padding(padding: const EdgeInsets.only(right: 12), child: card))).toList())
                  : Column(children: cards.map((card) => Padding(padding: const EdgeInsets.only(bottom: 12), child: card)).toList());
            },
          ),
          const SizedBox(height: 16),
          CalmCard(
            child: Text('Your notes helped ${stats.totalHelpfulNotes} people\n\nEven short words can help someone keep going.'),
          ),
          const SizedBox(height: 22),
          ListTile(
            shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(18)),
            tileColor: Theme.of(context).colorScheme.surface,
            leading: const Icon(Icons.bookmark_border),
            title: const Text('Saved notes'),
            subtitle: Text(repo.savedNotes.isEmpty ? 'No saved notes yet' : '${repo.savedNotes.length} saved'),
          ),
          const SizedBox(height: 10),
          ListTile(
            shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(18)),
            tileColor: Theme.of(context).colorScheme.surface,
            leading: const Icon(Icons.mail_outline),
            title: const Text('Report a concern'),
            subtitle: const Text('support@stillherehope.com'),
          ),
          const SizedBox(height: 10),
          ListTile(
            shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(18)),
            tileColor: Theme.of(context).colorScheme.surface,
            leading: const Icon(Icons.settings_outlined),
            title: const Text('Settings'),
            trailing: const Icon(Icons.chevron_right),
            onTap: () => context.push('/settings'),
          ),
          const SizedBox(height: 22),
          const SectionHeader(
            eyebrow: 'Patterns',
            title: 'Mood history',
            subtitle: 'A soft record of what days have asked from you.',
          ),
          const SizedBox(height: 12),
          if (repo.checkIns.isEmpty)
            const CalmCard(child: Text('No check-ins yet.'))
          else
            ...repo.checkIns.take(12).map(
                  (checkIn) => Padding(
                    padding: const EdgeInsets.only(bottom: 10),
                    child: CalmCard(
                      child: Text('${checkIn.mood.title}: ${checkIn.suggestedAction}'),
                    ),
                  ),
                ),
        ],
      ),
    );
  }
}

class _StatCard extends StatelessWidget {
  const _StatCard({required this.icon, required this.value, required this.label});
  final IconData icon;
  final String value;
  final String label;

  @override
  Widget build(BuildContext context) {
    return CalmCard(
      child: Column(
        crossAxisAlignment: CrossAxisAlignment.start,
        children: [
          Icon(icon, color: Theme.of(context).colorScheme.primary),
          const SizedBox(height: 14),
          Text(value, style: Theme.of(context).textTheme.headlineMedium?.copyWith(fontWeight: FontWeight.bold)),
          Text(label),
        ],
      ),
    );
  }
}
